### ISSUE TITLE
Bio::SeqIO Interproscan XML parsing issue

### ISSUE THREAD

**Issue by cjfields** [cjfields/bioperl-live-redmine#151]
---

Author Name: **Ben L** (Ben L)
Original Redmine Issue: 3452, https://redmine.open-bio.org/issues/3452
Original Date: 2013-11-08

---

Please see this post:

http://lists.open-bio.org/pipermail/bioperl-l/2013-November/071273.html

Thank you very much.

Ben

**Comment by cjfields** [comment]
---

Original Redmine Comment
Author Name: **Francisco J. Ossandon** 
Original Date: 2013-12-30T16:22:39Z

---

I wrote this by email but I will also leave it here for the record… In summary the parsing broke because there is a new Interproscan XML format. More below…

I’ve looked more into it and now that the match error is fixed, I can see that the reason that there is no data recovered from the parsing is because the XML structure is completely changed from what BioPerl is currently expecting.

This comes from the XML format change from the previous InterProScan 4 (https://www.ebi.ac.uk/Tools/pfa/iprscan/, the one recognized by BioPerl) and the current InterProScan 5 (https://www.ebi.ac.uk/interpro/, https://www.ebi.ac.uk/interpro/resources/schemas/interproscan5/). The new format looks more complex and have different type of matches (coils-match, fingerprints-match, hmmer2-match, hmmer3-match, panther-match, patternscan-match, etc.)

BioPerl looks for the structure “/protein/interpro/match” (InterProScan4):
@
 <protein id="Sequence_1" length="450" crc64="43336C02EEB5147C">
 <interpro id="IPR001957" name="Chromosomal replication control, initiator DnaA" type="Family" parent_id="IPR020591">
 <classification id="GO:0003677" class_type="GO">
 <match id="MF_00377" name="DnaA_bact" dbname="HAMAP">
 <location start="14" end="449" score="32.353" status="T" evidence="HAMAP"/>
@
But the new structure is “”/protein/matches/[different_types] " (InterProScan5):
@
 <protein>
 <sequence md5="934b35292f215a7e9b124c58a4cbb87d">MSSHSAPTALQDGAALWSALCVQLELVTSPQQFNTWLRPLRGELQGHELRLLAPNPFVRDWVRERMAELVKEQLQRIAPGFELVFALDEEAAAATSAPTASIAPERSSAPGGHRLNPAFNFQSYVEGKSNQLALAAARQVAQHPGKSYNPLYIYGGVGLGKTHLMQAVGNDILQRQPEAKVLYISSEGFIMDMVRSLQHNTINDFKQRYRKLDALLIDDIQFFAGKDRTQEEFFHTFNALFDGGRQIIITCDRYPKEVEGLEERLQSRFGWGLTVAIQPHDLETRMAIVLCKAEDHGIQLPEEVAFFIAEKIRSHVRELEGALRRVIAHVNFTHKPYSVESAKEALRDLIDVQKRMVSLENIQKVVADYYHIRASEMQSKRRNRNVARPRQMAMALTKELTRHSLPEIGEAFGGRDHTTVLHACRQIEKLRRESAQIEEDYRNLIRILGA</sequence>
 <xref name="sequence" id="Submitted"/>
 <matches>
 <hmmer2-match score="153.7" evalue="0.0">
 <signature name="Bac_DnaA_C" desc="Bacterial dnaA protein helix-turn-helix domain" ac="SM00760">
 <entry type="DOMAIN" name="DnaA_C" desc="Chromosomal replication initiator, DnaA C-terminal" ac="IPR013159">
 <go-xref category="BIOLOGICAL_PROCESS" name="DNA replication initiation" id="GO:0006270" db="GO"/>
 <go-xref category="BIOLOGICAL_PROCESS" name="regulation of DNA replication" id="GO:0006275" db="GO"/>
 <go-xref category="MOLECULAR_FUNCTION" name="sequence-specific DNA binding" id="GO:0043565" db="GO"/>
 <go-xref category="MOLECULAR_FUNCTION" name="ATP binding" id="GO:0005524" db="GO"/>
 </entry>
 <models>
 <model name="Bac_DnaA_C" desc="Bacterial dnaA protein helix-turn-helix domain" ac="SM00760"/>
 </models>
 <signature-library-release version="6.2" library="SMART"/>
 </signature>
 <locations>
 <hmmer2-location hmm-length="0" hmm-end="70" hmm-start="1" evalue="5.40001756E-44" score="0.0" end="427" start="358"/>
 </locations>
 </hmmer2-match>
@
####
I’m attaching the output given by the use of IPS4 and IPS5 using the same query sequence for comparison, the new output is much bigger. This means that new code will be needed to properly extract the data from the new format…

In summary, this case is more an enhancement work (support new IPS5 format) than a bug from the now older IPS4 format.

**Comment by cjfields** [comment]
---

Original Redmine Comment
Author Name: **Jason Stajich** 
Original Date: 2014-02-23T06:16:41Z

---

sounds like you need a version to parse new version but still maintain the old behavior.

**Comment by cjfields** [comment]
---

Original Redmine Comment
Author Name: **Francisco J. Ossandon** 
Original Date: 2014-03-04T22:23:27Z

---

The new parsing require more time than the one I can afford right now, so I leave open to anyone this development. =)

The XML files attached in this report can be used as start point to that end.
